### PR TITLE
HPCC4J-578 Migrate existing jirabot logic to new Jira environment

### PR DESCRIPTION
**Pull request by jpmcmu**
- Updated Jirabot to work with current Jira and new Cloud Jira

Signed-off-by: James McMullan James.McMullan@lexisnexis.com

## Type of change:
- [X] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [X] I have created a corresponding JIRA ticket for this submission
- [X] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [X] The change has been fully tested:
  - [ ] This change does not cause any existing JUnits to fail.
  - [ ] I have include JUnit coverage to test this change
  - [ ] I have performed system test and covered possible regressions and side effects.
- [X] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] This change fixes the problem, not just the symptom

### CODE REVIEW

**Comment by jpmcmu** [comment]
@rpastrana please review

File: .github/workflows/Jirabot.yml [hunk 1]
```diff
@@ -39,28 +39,20 @@ jobs:
...
-                elif statusName == 'Resolved':
-                    transition = 'Reopen Issue'
-                elif statusName == 'Closed':
-                    transition = 'Reopen Issue'
-                else:
-                    transition = ''
+                transition = transitionMap.get(statusName, None)
 
-                if transition != '':
+                if transition != None and transition != '':
```
Comment (rpastrana) [inline .github/workflows/Jirabot.yml:55]: it's hard to tell, how's the transaction == None or '' case handled?

**Review by rpastrana — APPROVED** [review]
@jpmcmu looks fine, just one question. If the empty transition map case is handled appropriately, let's go ahead and squash

File: .github/workflows/Jirabot.yml (hunk 1 above)
Comment (jpmcmu) [reply .github/workflows/Jirabot.yml:55]: If transition == None or '' that indicates we don't need to make any transitions.

**Comment by jpmcmu** [comment]
@rpastrana squashed

File: .github/workflows/Jirabot.yml (hunk 1 above)
Comment (rpastrana) [reply .github/workflows/Jirabot.yml:55]: I understand what the conditions mean, I asked how that case is handled.
if the new table is empty, does that issue get logged anywhere?

Comment (jpmcmu) [reply .github/workflows/Jirabot.yml:55]: Ah sorry I didn't understand your question. No, that issue wouldn't get logged anywhere, but if the table were empty or didn't have the specified status we would default to None. To your point it is probably a good idea to add a log message if we hit None, an empty string though should still be treated as a valid no-op

**Comment by jpmcmu** [comment]
@rpastrana Added an additional error message for the transition == None case

File: .github/workflows/Jirabot.yml [hunk 2]
```diff
@@ -122,24 +120,39 @@ jobs:
...
+                else:
+                    jiraUserEmail = jiraUser.emailAddress
+
+                print("Jira User Email:" + str(jiraUserEmail))
 
                 issue = jira.issue(issue_name)
                 result = 'Jirabot Action Result:\n'
 
+                transitionMap = json.loads(os.environ['JIRA_ISSUE_TRANSITION_MAP'])
+                if not isinstance(transitionMap, dict):
```
Comment (rpastrana) [inline .github/workflows/Jirabot.yml:148]: This should be logged.

Comment (jpmcmu) [reply .github/workflows/Jirabot.yml:148]: Added a log statement and squashed

**Pull request merged**